### PR TITLE
#27: Added withTolerance convenience method

### LINKED ISSUE #27: Add withTolerance convenience method
### Situation

Currently, we use the `cellMatcher` from CellMatcherFactory to test the floating point values with tolerance. As a tester, I want to have a `withTolerance` convenience method on a builder.

### Acceptance Criteria

- Added `withTolerance` method
- Added tests
- Updated user guide

### PR DESCRIPTION
**Pull request by morazow**
Fixes #27.

### CODE REVIEW

File: .settings/org.eclipse.core.resources.prefs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 eclipse.preferences.version=1
```
Comment (jakobbraun) [inline .settings/org.eclipse.core.resources.prefs:1]: you made these changes on purpose?

File: doc/changes/changes_1.3.1.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Matcher for SQL Result Sets 1.3.1, released 2021-??-??
```
Comment (jakobbraun) [inline doc/changes/changes_1.3.1.md:1]: mention this file in changes.md

File: src/main/java/com/exasol/matcher/ResultSetStructureMatcher.java [hunk 3]
```diff
@@ -244,6 +247,17 @@ public Builder row(final Object... cellValues) {
             return this;
         }
 
+        /**
+         * Adds a tolerance value for fuzzy matching floating point values.
+         *
+         * @param tolerance a tolerance value for matching floating point values
+         * @return {@code this} for fluent programming
+         */
+        public Builder withTolerance(final BigDecimal tolerance) {
```
Comment (jakobbraun) [inline src/main/java/com/exasol/matcher/ResultSetStructureMatcher.java:256]: When I once implemented this feature @redcatbear did not like it in the review.
If I remember correct his argument was that he wants different tolerances for each column and that for non nummeric column tolerance makes no sense.

I think, however it's quite usefull. Maybe we could rename this method to `withDefaultNumberTolerance` as a compromise?

**Review by jakobbraun — CHANGES_REQUESTED** [review]

File: .settings/org.eclipse.core.resources.prefs (hunk 1 above)
Comment (morazow) [reply .settings/org.eclipse.core.resources.prefs:1]: No, I thought these were introduced by project-keeper?
If it is not by it, then probably by the eclipse language server.

File: src/main/java/com/exasol/matcher/ResultSetStructureMatcher.java (hunk 3 above)
Comment (morazow) [reply src/main/java/com/exasol/matcher/ResultSetStructureMatcher.java:256]: Yes, but it does not work. It is sometimes very likely the default tolerance of zero works. 

For example, these fails:

```scala
.withInputValues[Float](Seq(2.71f, 3.14F, null.asInstanceOf[Float]))
      .assertResultSet(
        table() //  
          .row(2.71) //
          .row(3.14) //
          .row(0.0) //
          .matches()
      )
```

Even though input and expected values are same floating point values, in this case the read value has `2.71000000999`. Thus, adding tolerance with `10e-9` helps.

So adding cellMatcher for each is not very convenient.

The `withDefaultNumberTolerance` is fine with me.

Comment (redcatbear) [reply src/main/java/com/exasol/matcher/ResultSetStructureMatcher.java]: For fuzzy matching default tolerances are acceptable as long as they are documented properly in the handbook and are selected carefully on a per-type level.

File: .settings/org.eclipse.core.resources.prefs (hunk 1 above)
Comment (jakobbraun) [reply .settings/org.eclipse.core.resources.prefs]: PK does not touch these files...

Comment (jakobbraun) [reply .settings/org.eclipse.core.resources.prefs]: only jdt.core.prefs and jdt.ui.prefs

File: src/main/java/com/exasol/matcher/ResultSetStructureMatcher.java (hunk 3 above)
Comment (jakobbraun) [reply src/main/java/com/exasol/matcher/ResultSetStructureMatcher.java]: so maybe add a paragraph to the user_guide?

Comment (morazow) [reply src/main/java/com/exasol/matcher/ResultSetStructureMatcher.java]: Great, then lets add this method.

I have also updated the user guide with all the options.

Comment (morazow) [reply src/main/java/com/exasol/matcher/ResultSetStructureMatcher.java]: Please review again.

File: .settings/org.eclipse.core.resources.prefs (hunk 1 above)
Comment (morazow) [reply .settings/org.eclipse.core.resources.prefs]: Yeap, removed.

**Review by jakobbraun — APPROVED** [review]

**Review by jakobbraun — APPROVED** [review]

**Pull request merged**